bridge-token-factory/src/lib.rs

```diff
@@ -224,6 +291,44 @@ impl BridgeTokenFactory {
         self.tokens.iter().collect::<Vec<_>>()
     }
 
+     /// Finish updating token metadata once the proof was successfully validated. 
+     /// Can only be called by the contract itself.
+    pub fn finish_updating_metadata(
+        &mut self,
+        #[callback]
+        #[serializer(borsh)]
+        verification_success: bool,
+        #[serializer(borsh)] token: String,
+        #[serializer(borsh)] name: String,
+        #[serializer(borsh)] symbol: String,
+        #[serializer(borsh)] decimals: u8,
+        #[serializer(borsh)] timestamp: u64,
+        #[serializer(borsh)] proof: Proof,
+    ) {
+        assert_self();
+        assert!(verification_success, "Failed to verify the proof");
+        let required_deposit = self.record_proof(&proof);
+
+        assert!(
+            env::attached_deposit()
+                >= required_deposit + self.bridge_token_storage_deposit_required
+        );
+        env::log(format!("Finish updating metadata. Name:{} Symbol:{:?} Decimals:{:?} at : {:?}", name, symbol, decimals, timestamp).as_bytes());
```

Comment:
It's not clear what's `at` for without checking the sources, so either convert the timestamp to Human date or add the suffix telling that this is indeed a timestamp. IMO the second option is preferable as it doesn't require any additional computation.